src/kvstore/raftex/Host.cpp

```diff
@@ -280,7 +280,18 @@ Host::prepareAppendLogRequest() {
         << idStr_ << "My lastLogId in wal is " << part_->wal()->lastLogId()
         << ", but you are seeking " << lastLogIdSent_ + 1
         << ", so i have nothing to send, logIdToSend_ = " << logIdToSend_;
-    return nebula::cpp2::ErrorCode::E_RAFT_NO_WAL_FOUND;
+    auto req = std::make_shared<cpp2::AppendLogRequest>();
+    req->space_ref() = part_->spaceId();
+    req->part_ref() = part_->partitionId();
+    req->current_term_ref() = logTermToSend_;
+    req->committed_log_id_ref() = committedLogId_;
+    req->leader_addr_ref() = part_->address().host;
+    req->leader_port_ref() = part_->address().port;
+    req->last_log_term_sent_ref() = lastLogTermSent_;
+    // the purpose of setting last_log_id to -1 is to make remote host return committedLogId which
+    // could be used as lastLogIdSent in next loop
+    req->last_log_id_sent_ref() = -1;
+    return req;
```

Comment:
Em... I don't understand why the change in the PR could solve `lastLogIdSent_ + 1 > part_->wal()->lastLogId()`?

Even if the peer return `committedLogId`, but leader still doesn't have the wal, won't it still loop forever?

```
send a request with -1 
                                          -> return committedLogId
send a request with -1 
                                          -> return committedLogId
...
```

Reply:
yes, only when when committedLogId == part_->wal()->lastLogId() could lead this condition.
just set req->last_log_id_sent_ref() = lastLogIdSent_ could solve the problem.